src/Merchants.php

```diff
@@ -142,15 +142,36 @@ public static function update_or_create_merchant() {
 			'merchant_name'    => $merchant_name,
 		);
 
-		// The response only contains the merchant id.
-		$response = API\Base::update_or_create_merchant( $args );
+		$cache_key = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '_request_' . md5( wp_json_encode( $args ) );
+		$cache     = get_transient( $cache_key );
+
+		if ( false !== $cache ) {
+			throw new Exception( __( 'There was a previous error trying to create or update merchant.', 'pinterest-for-woocommerce' ), (int) $cache );
+		}
+
+		try {
+			// The response only contains the merchant id.
+			$response = Base::update_or_create_merchant( $args );
+		} catch ( Throwable $th ) {
+			$delay = Pinterest_For_Woocommerce()::get_data( 'create_merchant_delay' ) ?? 10 * MINUTE_IN_SECONDS;
```

Comment:
I did tested it and it works. Though I think that 10 minutes is much too aggressive. I have counted a multiple retries in a matter of couple of seconds when I have disabled the delay, if the first one would be just a fluke then the whole site would be blocked for the 10 minutes. I think that we can start with 1 minute. Getting to 10 is just 4 failed retries.

Reply:
Done :white_check_mark: